### PR TITLE
fix: Currency amount significant digits

### PR DESCRIPTION
**Pull request by memoyil**
- Go to remove currency v3 with a token amount between 0 - 1
- See it is shown as 0

### Summary
🔧🚚📝

This pull request refactors the `FormattedCurrencyAmount` component and fixes some type and import issues in the liquidity views. It also allows `undefined` values for some of the localization context data. These changes improve the code quality and functionality of the web app.

> _We're breaking free from the liquidity pools_
> _We're fixing types and imports, we're not fools_
> _We're moving components to a common place_
> _We're formatting numbers with style and grace_

### Walkthrough
*  Change the import path for `FormattedCurrencyAmount` in four files to reflect its new location (`[link](https://github.com/pancakeswap/pancake-frontend/pull/8299/files?diff=unified&w=0#diff-d541d696a78f40e25e2fb5e5596eee09f23c12573229e2734b5bdfc1ef40b22bL43-R43)`, `[link](https://github.com/pancakeswap/pancake-frontend/pull/8299/files?diff=unified&w=0#diff-7c64b75d57438b6b758bfaf8af6c56ebcedf283b93fc2a606c5b4713fa902724L25-R29)`, `[link](https://github.com/pancakeswap/pancake-frontend/pull/8299/files?diff=unified&w=0#diff-855ff4967fce38b55c660fd0a23d5093b024f95f1ce41fa5c94b3a705ed13cd3L16-R16)`, `[link](https://github.com/pancakeswap/pancake-frontend/pull/8299/files?diff=unified&w=0#diff-3639b5beb9d763170cb423e83fcbcd8129c820faa28a757bf95b2d841ca14261L43-R43)`)

### CODE REVIEW

File: apps/web/src/pages/liquidity/[tokenId].tsx [hunk 1]
```diff
@@ -509,10 +509,12 @@ export default function PoolPage() {
                     </Text>
                     {isMobile && (
                       <Flex>
-                        {isStakedInMCv3 && (
+                        {isStakedInMCv3 ? (
                           <Tag mr="8px" outline variant="warning">
                             {t('Farming')}
                           </Tag>
+                        ) : (
+                          <></>
```
Comment (chefjackson) [inline apps/web/src/pages/liquidity/[tokenId].tsx:517]: we can use `null` instead

File: apps/web/src/pages/liquidity/[tokenId].tsx [hunk 2]
```diff
@@ -361,9 +361,9 @@ export default function PoolPage() {
   ])
 
   const owner = useSingleCallResult({
-    contract: tokenId ? positionManager : null,
+    contract: tokenId && positionManager ? positionManager : undefined,
     functionName: 'ownerOf',
-    args: useMemo(() => [tokenId] as const, [tokenId]),
+    args: useMemo(() => (tokenId ? [tokenId] : []) as [bigint], [tokenId]),
```
Comment (chefjackson) [inline apps/web/src/pages/liquidity/[tokenId].tsx:366]: if we're fixing typing issue, can we not change the original behavior and simply add the type assertion?
```suggestion
    args: useMemo(() => [tokenId] as [bigint], [tokenId]),
```

File: apps/web/src/pages/liquidity/[tokenId].tsx [hunk 3]
```diff
@@ -990,7 +989,7 @@ function PositionHistoryRow({
   currency0,
   currency1,
 }: {
-  chainId: ChainId
+  chainId: ChainId | undefined
```
Comment (chefjackson) [inline apps/web/src/pages/liquidity/[tokenId].tsx:992]: just a suggestion, if it's possibly undefined, we can use the optional mark
```suggestion
  chainId?: ChainId
```

File: apps/web/src/state/multicall/hooks.ts [hunk 4]
```diff
@@ -235,16 +235,19 @@ export function useSingleContractMultipleData<TAbi extends Abi | readonly unknow
...
+              return {
+                address: contract.address,
+                callData: encodeFunctionData({
+                  abi: contract.abi,
+                  functionName,
+                  args: inputs,
+                } as unknown as EncodeFunctionDataParameters),
+              }
+            })
+            .filter(Boolean)
```
Comment (chefjackson) [inline apps/web/src/state/multicall/hooks.ts:250]: Do we need to make sure the returned type is `Call[]` here?

File: apps/web/src/views/AddLiquidityV3/components/AprCalculator.tsx [hunk 5]
```diff
@@ -114,15 +114,21 @@ export function AprCalculator({
     () =>
       baseUSDPrice
         ? parseFloat(formatPrice(baseUSDPrice, 6) || '0')
-        : deriveUSDPrice(quoteUSDPrice, price?.baseCurrency.equals(quoteCurrency?.wrapped) ? price : price?.invert()),
-    [baseUSDPrice, quoteUSDPrice, price, quoteCurrency?.wrapped],
+        : deriveUSDPrice(
+            quoteUSDPrice,
+            quoteCurrency && price?.baseCurrency.equals(quoteCurrency?.wrapped) ? price : price?.invert(),
```
Comment (chefjackson) [inline apps/web/src/views/AddLiquidityV3/components/AprCalculator.tsx:119]:
```suggestion
            quoteCurrency && price?.baseCurrency.equals(quoteCurrency.wrapped) ? price : price?.invert(),
```

File: apps/web/src/views/AddLiquidityV3/components/AprCalculator.tsx [hunk 6]
```diff
@@ -114,15 +114,21 @@ export function AprCalculator({
...
+          ),
+    [baseUSDPrice, quoteUSDPrice, price, quoteCurrency],
   )
   const currencyBUsdPrice = useMemo(
     () =>
       baseUSDPrice &&
-      (deriveUSDPrice(baseUSDPrice, price?.baseCurrency.equals(baseCurrency?.wrapped) ? price : price?.invert()) ||
+      (deriveUSDPrice(
+        baseUSDPrice,
+        baseCurrency && price?.baseCurrency.equals(baseCurrency?.wrapped) ? price : price?.invert(),
```
Comment (chefjackson) [inline apps/web/src/views/AddLiquidityV3/components/AprCalculator.tsx:128]:
```suggestion
        baseCurrency && price?.baseCurrency.equals(baseCurrency.wrapped) ? price : price?.invert(),
```

File: apps/web/src/state/multicall/hooks.ts [hunk 7]
```diff
@@ -235,7 +235,8 @@ export function useSingleContractMultipleData<TAbi extends Abi | readonly unknow
   const calls = useMemo(
     () =>
       contract && contract.abi && contract.address && args && args.length > 0
-        ? args.map<Call>((inputs) => {
+        ? args.map((inputs) => {
+            if (!contract.address) return undefined
```
Comment (chefjackson) [inline apps/web/src/state/multicall/hooks.ts:239]: I think `contract.address` can't be nullish? As we have checked it in line 237

File: apps/web/src/state/multicall/hooks.ts [hunk 8]
```diff
@@ -235,7 +235,8 @@ export function useSingleContractMultipleData<TAbi extends Abi | readonly unknow
   const calls = useMemo(
     () =>
       contract && contract.abi && contract.address && args && args.length > 0
-        ? args.map<Call>((inputs) => {
+        ? args.map((inputs) => {
```
Comment (chefjackson) [inline apps/web/src/state/multicall/hooks.ts:238]: Do we need to add type assertion `as Call[]` at the end of the map call if we remove the generic `<Call>` here?

**Review by MustardseedX — COMMENTED** [review]
Q

**Review by chefjackson — APPROVED** [review]

**Pull request merged**